docs/guides/making_HUD_using_ui-component.rst

```diff
@@ -0,0 +1,107 @@
+=============================
+Making HUD using UI-component
+=============================
+
+UI creation is quite easy in Rootex. The UI-component spans the viewport and is not affected by the camera transform, so it can act be used for creating HUD, Menus etc.
+
+RmlUi
+=====
+
+The UI-component makes use of RmlUi. RmlUi is a C++ UI library based on the HTML and CSS standards.
+It contains of **RML** (based loosely around XHTML 1.0 and HTML 4.01) and **RCSS** (based on CSS2).
+`RmlUi Documentation <https://mikke89.github.io/RmlUiDoc/>`_
+`RmlUi Source <https://github.com/mikke89/RmlUi>`_
+
+Basic HUD
+=========
+
+Let's create an empty scene.
+
+Add a UIComponent to this scene
+
+Make an empty scene with a TransformComponent and ModelComponent (basic cube)
+
+.. image::images/uidemo_1.png
+   :scale: 50
+
+Now add the following RML script to the UIComponent
+
+``<rml>
+	<head>
+		<title>Demo</title>
+		<style>
+			#transition_test {
+				transition: padding-left background-color transform 1.6s elastic-out;
+				transform: scale(1.0);
+				background-color: #c66;
+			}
+			#transition_test:hover {
+				padding-left: 60px;
+				transform: scale(1.5);
+				background-color: #f8b600;
+			}
+			p
+			{
+				font-family: "Lato";
+				font-size: 100px;
+				color: purple;
+				font-effect: glow(2px 4px 2px 3px #644);
+				border: 2px #DB6565;
+			}
+            body
+			{
+				width: 100%;
+				height: 100%;
+				margin: auto;
+			}
+        </style>
+    </head>
+    <body>
+		<p id="transition_test">This is a sample!</p>
+    </body>
+</rml>``
```

Comment:
This is getting rendered as plain text instead of a code snippet.

![image](https://user-images.githubusercontent.com/58972469/155040231-abc00d95-7b05-4a8a-8f53-917b059c9fd3.png)